File: core/src/main/java/com/novoda/materialised/MessagePresenter.java
package com.novoda.materialised;

public class MessagePresenter {
    private final ToggleMessages toggleMessages;
    private boolean isToggled = false;

    public MessagePresenter(ToggleMessages toggleMessages) {
        this.toggleMessages = toggleMessages;
    }

    private boolean shouldShowDefaultMessage() {
        Boolean isCurrentlyToggled = isToggled;

        isToggled = !isToggled;

        return isCurrentlyToggled;
    }

    public String currentMessage() {
        return shouldShowDefaultMessage() ? toggleMessages.getDefaultMessage() : toggleMessages.getToggledMessage();
    }
}

File: core/src/main/java/com/novoda/materialised/example/MessagePresenter.kt
package com.novoda.materialised.example

class MessagePresenter(toggleMessages: ToggleMessages) {
    private val toggleMessages: ToggleMessages = toggleMessages
    private var isToggled = false

    fun currentMessage(): String {
        return if (shouldShowDefaultMessage()) toggleMessages.defaultMessage else toggleMessages.toggledMessage
    }

    private fun shouldShowDefaultMessage(): Boolean {
        val isCurrentlyToggled = isToggled

        isToggled = !isToggled

        return isCurrentlyToggled
    }
}
